r-lib-error-1.0.md

```diff
@@ -0,0 +1,64 @@
+
+# Specification of r-lib errors, version 1.0.0
+
+Note: this is an experimental draft.
+
+Note: version 1.0.0 corresponds to the `rlang_error` class, as
+implemented in rlang 0.4.0.
+
+## Classes
+
+An r-lib error object always has at least classes `rlib_error`, `error`
+and `condition`. It may have other classes.
+
+## Members
+
+An `rlib_error` object has the following members:
+
+* `message`: `character(1)`, optional, the error message. If this is
```

Comment:
Thinking more about it, I don't think this belongs in the spec, as this is all implied by inheriting from `condition`. The `condition` methods for `conditionMessage()` and `conditionCall()` methods look into these fields, this is how they are part of the API.

Reply:
That's a contract for `condition`, but it is entirely possible to strengthen that contract and require a `message` field. So I think it makes sense to mention it, especially since most `rlib_error` objects will have a `message` field, even if we don't require it.

Reply:
I agree that we can, but why should we? This seems out of place to me.

Reply:
I think it's at important to note the contract we believe we inherit from condition as that's not otherwise documented.